### DISCUSSION TITLE
“Award Criteria Complicated” (BT-543-Lot) in CAN 29

### DISCUSSION THREAD

**Discussion by lesomue** [OP-TED/eForms-SDK#922] [Q&A] [answered]
Hello everyone, 

During the implementation of SDK 1.10.1 in our system, we noticed during tests that it is apparently not possible to enter "Award Criteria Complicated" (BT-543-Lot) in a contract award notice (CAN 29) without entering one of the award criteria numbers (BT-541-Lot-FixedNumber, BT-541-Lot-ThresholdNumber, BT-541-Lot-WeightNumber). 

If, for example, award criteria are used in Form 29 that cannot be expressed using a standard weighting (i.e. using BT-543-Lot), the validation errors BT-00541-0303, BT-00541-0503 and BR-BT-00541-0703 occur, but only if unpublishers are also used for the award criteria. Without unpublishers for the award criteria, the rules do not apply and no error message is displayed. This raises the following questions for us: 

1. How exactly may award criteria be included in contract notice no. 29? It seems that BT-543 should not be used in a contract award notice, as this information is forbidden if no award criteria number (BT-541) is included. However, a numer must be recorded in the CAN if a description (BT-540) is entered, which in turn must be entered if an award criterion type (BT-539) is used, which in turn is always mandatory in notice no. 29. On the other hand, it is stipulated for BT-543 in fields.json that the information on BT-543 is always mandatory if it is not forbidden, although it is always forbidden by tracing down the validation chain. We would like to ask for your feedback on this matter and how to proceed in this case. 
2. Why are BT-00541-0303, BT-00541-0503 and BR-BT-00541-0703 only triggered if unpublishers are also used? 

Many thanks in advance and best regards
Lea Müller
cosinex GmbH

**Comment by YvesJo** [comment, marked as answer]
Hi,

> During the implementation of SDK 1.10.1 in our system, we noticed during tests that it is apparently not possible to enter "Award Criteria Complicated" (BT-543-Lot) in a contract award notice (CAN 29) without entering one of the award criteria numbers (BT-541-Lot-FixedNumber, BT-541-Lot-ThresholdNumber, BT-541-Lot-WeightNumber).

BT-543-Lot is forbidden as soon as BT-541-Lot-WeightNumber, BT-541-Lot-FixedNumber or BT-541-Lot-ThresholdNumber exists (Stage 4 Conditionally Forbidden rules on BT-543-Lot, BR-BT-00543-0106 for Notice Subtype 29)
For the specific case of Notice Subtype 29, there is also a Stage 4 Conditionally Mandatory rule (BR-BT-00543-0088) making BT-543-Lot mandatory when not forbidden (i.e. when BT-541-Lot-WeightNumber, BT-541-Lot-FixedNumber and BT-541-Lot-ThresholdNumber do not exist).

> It seems that BT-543 should not be used in a contract award notice, as this information is forbidden if no award criteria number (BT-541) is included.

Incorrect conclusion, see above.
More generally:
- BT-539-Lot is Mandatory without condition (stage 2) for Notice Subtype 29
- BT-540-Lot is forbidden when BT-539-Lot does not exist (BR-BT-00540-0211) and mandatory as soon as it does (BR-BT-00540-0088).

Hence BT-539-Lot and BT-540-Lot must exist for Notice Subtype 29

The node that contains all the criteria parameters (ND-LotAwardCriterionParameters) is the context that triggers the following rules, which in its absence do not fire. Also this node should not exist in the XML if empty.
- BT-541-Lot-xxx are forbidden as long as no description is provided (BR-BT-00541-0285, BR-BT-00541-0485, BR-BT-00541-0685). 
- At least one of the three BT-541-Lot- xxx has to exist (BR-BT-00541-0303, BR-BT-00541-0503, BR-BT-00541-0703) 

Please note that the unpublish fields related to the parameters are also under ND-LotAwardCriterionParameters, therefore, if they exist, the node ND-LotAwardCriterionParameters will also exist and the rules get fired.

**Comment by lesomue** [comment]
Many thanks for your reply. 

Just to be clear: 
- BT-543 is forbidden, if BT-541 exists. If not, BT-543 is mandatory.
- BT-549 and BT-540 are always mandatory for notice subtype 29. 
- If BT-540 exists, BT-541 is mandatory. 
- Since BT-540 is mandatory in 29, BT-541 should always exist according to fields.json. 
- Due to the design of the validation rules, "At least one of the three BT-541-Lot- xxx has to exist (BR-BT-00541-0303, BR-BT-00541-0503, BR-BT-00541-0703)" is not triggered, as the context node is not existing in the case of BT-543. 
- The context node for BR-BT-00541-0303, BR-BT-00541-0503, BR-BT-00541-0703 only exists, if there are unpublish fields for BT-539 for expample and then the rules are triggered. 

Summarizing that, I conclude that contracting authorities can either choose 
**a.** to use BT-543 if the award criteria is too complex to be expressed by parameters and cannot choose to use unpublish fields for the award criteria. 
**b.** to use unpublish fields but not BT-543 and instead use fictitious parameters (BT-541) which do not depict reality. 

Please let me know, if my understanding is correct.

**Reply by YvesJo** [reply]
> - BT-543 is forbidden, if BT-541 exists. If not, BT-543 is mandatory.

This is exact.

> - BT-549 and BT-540 are always mandatory for notice subtype 29.

Yes, BT-539 and BT-540 are always required for Notice Subtype 29

> - If BT-540 exists, BT-541 is mandatory.

This is only true when the node that triggers that rule exists (i.e. any of its descendants already exist, e.g. BT-541 or BT-195(BT-541) ... ) 

> - Since BT-540 is mandatory in 29, BT-541 should always exist according to fields.json.

No, because the node triggering the rule may not exist, you may find it as part of the condition ("condition" : "**{ND-LotAwardCriterionParameters}** ...)

> - Due to the design of the validation rules, "At least one of the three BT-541-Lot- xxx has to exist (BR-BT-00541-0303, BR-BT-00541-0503, BR-BT-00541-0703)" is not triggered, as the context node is not existing in the case of BT-543.

This is exact

> - The context node for BR-BT-00541-0303, BR-BT-00541-0503, BR-BT-00541-0703 only exists, if there are unpublish fields for BT-539 for expample and then the rules are triggered.

ThIs is unfortunately the case as the unpublish fields for the Criterion Name and Description share the same node (ND-LotAwardCriterionParameters) and the condition for the rules BR-BT-00541-0303, BR-BT-00541-0503, BR-BT-00541-0703 should be updated to exclude the situation where BT-543 would be specified.
This will get updated in SDK 1.12 and backported after.

**Comment by lesomue** [comment]
Okay, many thanks for your clarification!
